### ISSUE TITLE
Write access is allowed from write-safe contexts only.

### ISSUE THREAD

**Issue by rnentjes** [ebean-orm/ebean-idea-enhancer#13]
Hi, 

Fyi; I got the error message below with the enhancer in intellij:

Cheers, Rien

```
Write access is allowed from write-safe contexts only. Please ensure you're using invokeLater/invokeAndWait with a correct modality state (not "any"). See TransactionGuard documentation for details.
  current modality=ModalityState.NON_MODAL
  known modalities={ModalityState:Commit dialog: Local Changes Refresh 208509115: running=false; canceled=false=true, ModalityState.NON_MODAL=true, ModalityState:com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog[dialog1,2302,66,922x947,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=Commit Changes,defaultCloseOperation=DO_NOTHING_ON_CLOSE,rootPane=,rootPaneCheckingEnabled=true]=true, ModalityState:com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog[dialog1,2302,66,922x947,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=Commit Changes,defaultCloseOperation=DO_NOTHING_ON_CLOSE,rootPane=,rootPaneCheckingEnabled=true], com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog[dialog3,2103,104,1554x870,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=TestCurrenciesService.java (C:\Development\peachtree\backend\src\test\java\com\banktrade\clienttrade\admin\module),defaultCloseOperation=DO_NOTHING_ON_CLOSE,rootPane=,rootPaneCheckingEnabled=true]=true, ModalityState:com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog[dialog1,2302,66,922x947,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=Commit Changes,defaultCloseOperation=DO_NOTHING_ON_CLOSE,rootPane=,rootPaneCheckingEnabled=true]=true, ModalityState:com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog[dialog1,2302,66,922x947,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=Commit Changes,defaultCloseOperation=DO_NOTHING_ON_CLOSE,rootPane=,rootPaneCheckingEnabled=true]=true, ModalityState:com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog[dialog1,2302,66,922x947,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=Commit Changes,defaultCloseOperation=DO_NOTHING_ON_CLOSE,rootPane=,rootPaneCheckingEnabled=true]=true, ModalityState:com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog[dialog1,2302,66,922x947,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=Commit Changes,defaultCloseOperation=DO_NOTHING_ON_CLOSE,rootPane=,rootPaneCheckingEnabled=true]=true, ModalityState:com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog[dialog1,2302,66,922x947,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=Commit Changes,defaultCloseOperation=DO_NOTHING_ON_CLOSE,rootPane=,rootPaneCheckingEnabled=true]=true}
java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:132)
	at com.intellij.openapi.application.TransactionGuardImpl.assertWriteActionAllowed(TransactionGuardImpl.java:246)
	at com.intellij.openapi.vfs.newvfs.RefreshQueueImpl.execute(RefreshQueueImpl.java:54)
	at com.intellij.openapi.vfs.newvfs.RefreshSessionImpl.launch(RefreshSessionImpl.java:115)
	at com.intellij.openapi.vfs.newvfs.RefreshQueueImpl.processSingleEvent(RefreshQueueImpl.java:131)
	at com.intellij.openapi.vfs.newvfs.impl.VirtualDirectoryImpl.a(VirtualDirectoryImpl.java:257)
	at com.intellij.openapi.vfs.newvfs.impl.VirtualDirectoryImpl.a(VirtualDirectoryImpl.java:99)
	at com.intellij.openapi.vfs.newvfs.impl.VirtualDirectoryImpl.refreshAndFindChild(VirtualDirectoryImpl.java:264)
	at com.intellij.openapi.vfs.newvfs.VfsImplUtil.refreshAndFindFileByPath(VfsImplUtil.java:137)
	at com.intellij.openapi.vfs.impl.local.LocalFileSystemBase.refreshAndFindFileByPath(LocalFileSystemBase.java:74)
	at com.intellij.openapi.vfs.impl.local.LocalFileSystemBase.refreshAndFindFileByIoFile(LocalFileSystemBase.java:229)
	at com.intellij.openapi.vfs.VfsUtil.findFileByIoFile(VfsUtil.java:234)
	at org.avaje.idea.ebean10.plugin.EbeanEnhancementTask.writeTransformed(EbeanEnhancementTask.java:195)
	at org.avaje.idea.ebean10.plugin.EbeanEnhancementTask.processEnhancement(EbeanEnhancementTask.java:149)
	at org.avaje.idea.ebean10.plugin.EbeanEnhancementTask.lambda$doProcess$1(EbeanEnhancementTask.java:137)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:934)
	at org.avaje.idea.ebean10.plugin.EbeanEnhancementTask.doProcess(EbeanEnhancementTask.java:136)
	at com.intellij.util.ActionRunner$1.run(ActionRunner.java:31)
	at com.intellij.openapi.application.RunResult.run(RunResult.java:35)
	at com.intellij.openapi.application.WriteAction.execute(WriteAction.java:38)
	at com.intellij.util.ActionRunner.runInsideWriteAction(ActionRunner.java:33)
	at org.avaje.idea.ebean10.plugin.EbeanEnhancementTask.process(EbeanEnhancementTask.java:74)
	at org.avaje.idea.ebean10.plugin.CompiledFileCollector.compilationFinished(CompiledFileCollector.java:94)
	at sun.reflect.GeneratedMethodAccessor787.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.util.messages.impl.MessageBusConnectionImpl.deliverMessage(MessageBusConnectionImpl.java:117)
	at com.intellij.util.messages.impl.MessageBusImpl.doPumpMessages(MessageBusImpl.java:381)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:368)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:354)
	at com.intellij.util.messages.impl.MessageBusImpl.sendMessage(MessageBusImpl.java:347)
	at com.intellij.util.messages.impl.MessageBusImpl.access$200(MessageBusImpl.java:42)
	at com.intellij.util.messages.impl.MessageBusImpl$2.invoke(MessageBusImpl.java:230)
	at com.sun.proxy.$Proxy77.compilationFinished(Unknown Source)
	at com.intellij.compiler.CompilerManagerImpl$ListenerNotificator.finished(CompilerManagerImpl.java:496)
	at com.intellij.compiler.impl.CompileDriver.a(CompileDriver.java:486)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at com.intellij.ide.IdeEventQueue.b(IdeEventQueue.java:843)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:679)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:391)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by vnnv** [comment]
I notice this exception too

**Comment by rbygrave** [comment]
Fixed with 10.1.6 version of this plugin.  Just uploaded to jetbrains repo and .zip also pushed to this git repo.
